packages/eslint-plugin/src/util/getThisExpression.ts

```diff
@@ -0,0 +1,26 @@
+import {
+  AST_NODE_TYPES,
+  TSESTree,
+} from '@typescript-eslint/experimental-utils';
+
+export function getThisExpression(
```

Comment:
one case this doesn't handle is short-circuited optional chains which use the `ChainExpression` node
`(this.x?.y).z`

Reply:
Done!